### DISCUSSION TITLE
左右自動判定のための追加工作について推奨方法を決めたい

### DISCUSSION THREAD

**Discussion by koron** [Yowkees/keyball#24] [Ideas]
左右自動判定をする際には左側のボードの `SW` をショートさせるのを推奨、という方針にしたいです。

これにより左右自動判定できるビルド済みファームウェアを提供し、ユーザーにとって使いやすくするのが目的です。

関連: https://github.com/Yowkees/keyball/discussions/16

## 背景

現在のsplit keyboardではUSBケーブルをつないだ方を右か左どちらかに固定する、という習慣があります。ただこれには普段USBを繋がない方をテストするときや、片方だけで使いたいときに左右が反転したキーマップになり使いにくいという問題もあります。

Keyball46もその習慣に従って同じ問題を抱えていますが、加えて特有の事情としてトラックボールのある方にUSBを繋がなければならないという制約があり、事態を複雑にしています。

一方でKeball46には左右の自動判定に使える、未利用のmatrix intersectionが存在し、パッドも `SW` として明示されているという有利な点がありこれを利用しないのはもったいないと言えます。ただ利用する際に人によって追加加工するのが左だったり右だったりすると、前述のKeyball固有の事情もあり難しくなりすぎるのは良くありません。

なのでそれらを単純化するために「(オプション)左右自動判定をしたい場合は左側の `SW` をショートさせる」という風にビルドガイドに書いてあれば、そのようにして設定してビルドしたファームを用意しておくとユーザー体験が良くなると考えます。

ちなみに追加加工を左側としたのは、決めだけの問題なので右側でも良いのですが、右側トラックボールを選択する人が多いと予想されることから、あとから追加加工しやすいのはトラックボール無しの左側になるだろう、という意図です。

**Comment by Yowkees** [comment]
ご提案ありがとうございます！いつもご尽力いただいて本当に感謝です。
ラッキーにも回路構成もマッチしており、ぜひ左右判別機能は搭載してみたいと思っておりました。

ぜひビルドガイドに追記したいと思います。
写真撮影がありますので、日曜日には反映いたします。

シングルボール版でもダブルボール版でも、必ず左手側のみ”SW”をショートさせるという理解ですがよろしいでしょうか？
また本作業は必須でも良いかと思いますが、ショートさせないほうが有利な場面というのはあるでしょうか？（SWが一つ拡張できるというのは、無しで良いかと思っています）

**Reply by koron** [reply]
> 必ず左手側のみ”SW”をショートさせるという理解ですがよろしいでしょうか？

はい。あっています。

> ショートさせないほうが有利な場面というのはあるでしょうか？

いいえ。考えうる限り有利な場面は思いつきません。

仮に問題があるとすれば、
まだ僕の手元でも加工していない
ファームを用意できていない、ということでしょうか。

そのあたりがクリアできれば良いかと思います。
土曜には全部試せるとは思います。

**Comment by koron** [comment]
Keyball46 のファームウェアはこの追加工作が前提になりました。

**Reply by Yowkees** [reply]
了解しました。ビルドガイドでは標準となっていますが、発売初期に買われた方にはアナウンスしようと思います。ありがとうございます。
